### PR TITLE
Exporter tests

### PR DESCRIPTION
**Pull request by XrayT3**

### CODE REVIEW

**Review by F-I-D-O — CHANGES_REQUESTED** [review]
I think we need more here. The map functions' tests only the get_map function with mocked data. But I would expect the greatest potential for errors in the get_nodes/get_edges functions and during the communication with the database. I suggest adding tests that work with the database, i.e., fill test db with sample node/edge data, and then test it as it is, no mocking.

Additionaly, you should expand the testing section of the readme with instruction how to run the exporter tests

**Review by F-I-D-O — CHANGES_REQUESTED** [review]

File: python/roadgraphtool/db.py [hunk 1]
```diff
@@ -150,6 +150,19 @@ def execute_sql(self, query, *args, use_transactions=True) -> None:
...
+        Execute SQL in a specific schema.
+        """
+        with self._sqlalchemy_engine.connect() as connection:
+            if not use_transactions:
+                connection.execution_options(isolation_level="AUTOCOMMIT")
+            with connection.begin():
+                connection.execute(sqlalchemy.text(f"SET search_path TO {schema};"))
+                connection.execute(sqlalchemy.text(query), *args)
+                connection.execute(sqlalchemy.text(f"SET search_path TO public;"))
+
```
[the remark this answers is not in the document]
Comment (XrayT3) [reply python/roadgraphtool/db.py:165]: I had the same idea, but I was affraid to break something. I will try and test this approach.

File: python/roadgraphtool/db.py [hunk 2]
```diff
@@ -140,15 +140,17 @@ def get_new_psycopg2_connection(self):
...
         """
         Execute SQL that doesn't return any value.
         """
         with self._sqlalchemy_engine.connect() as connection:
             if not use_transactions:
                 connection.execution_options(isolation_level="AUTOCOMMIT")
             with connection.begin():
+                connection.execute(sqlalchemy.text(f"SET search_path TO {schema};"))
                 connection.execute(sqlalchemy.text(query), *args)
+                connection.execute(sqlalchemy.text(f"SET search_path TO public;"))
```
Comment (F-I-D-O) [inline python/roadgraphtool/db.py:153]: this does not seem right :)

File: python/roadgraphtool/export.py [hunk 3]
```diff
@@ -33,9 +35,11 @@ def get_map_nodes_from_db(area_id: int) -> gpd.GeoDataFrame:
     return db.execute_query_to_geopandas(sql)
```
Comment (F-I-D-O) [inline python/roadgraphtool/export.py:35]: why the schema cannot be handled in `execute_query_to_geopandas`?

**Review by F-I-D-O — CHANGES_REQUESTED** [review]

File: python/roadgraphtool/db.py (hunk 2 above)
Comment (zlochina) [reply python/roadgraphtool/db.py:153]: You can save starting value of search_path by executing ‘search_path_tmp = db.execute_sql_and_fetch_all_rows(“show search_path;”)[0][0]’

Comment (F-I-D-O) [reply python/roadgraphtool/db.py:153]: no, I mean the path hardcoded to the public scheme? Is that correct?

**Review by F-I-D-O — APPROVED** [review]

**Pull request merged**